src/core/buffer.c

```diff
@@ -23,11 +22,8 @@ void buffer_set(buffer *buf, int offset)
 
 static int check_size(buffer *buf, int size)
 {
-    if (buf->index + size > buf->size) {
-        buf->overflow = 1;
-        return 0;
-    }
-    return 1;
+    buf->overflow = (buf->index + size) > buf->size;
```

Comment:
To be honest I find the original code cleaner: only setting the overflow variable iff there actually is an overflow.
Also, which value is stored in overflow is compiler dependent, so I try to avoid it.